src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/SubscriptionAT.java

```diff
@@ -71,6 +77,53 @@ public void testSubscriptionCreation() throws IOException {
         assertThat(subSecond, equalTo(subFirst));
     }
 
+    @Test
+    public void testOffsetsCommit() throws Exception {
+        // create event type in Nakadi
+        final EventType eventType = createEventType();
+
+        // create subscription
+        final String subscriptionJson = "{\"owning_application\":\"app\",\"event_types\":[\"" + eventType.getName() + "\"]}";
+        Response response = given()
+                .body(subscriptionJson)
+                .contentType(JSON)
+                .post(SUBSCRIPTIONS_URL);
+        final Subscription subscription = mapper.readValue(response.print(), Subscription.class);
+
+        // commit offsets and expect 200
+        given()
+                .body("[{\"partition\":\"0\",\"offset\":\"25\"}]")
+                .contentType(JSON)
+                .put(format(CURSORS_URL, subscription.getId()))
+                .then()
+                .statusCode(HttpStatus.SC_OK);
+
+        // check that offset is actually committed to Zookeeper
+        final CuratorFramework curator = ZooKeeperUtils.createCurator(zookeeperUrl);
+        String committedOffset = getCommittedOffsetFromZk(eventType, subscription, "0", curator);
+        assertThat(committedOffset, equalTo("25"));
+
+        // commit lower offsets and expect 204
+        given()
+                .body("[{\"partition\":\"0\",\"offset\":\"10\"}]")
+                .contentType(JSON)
+                .put(format(CURSORS_URL, subscription.getId()))
+                .then()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+
+        // check that committed offset in Zookeeper is not changed
+        committedOffset = getCommittedOffsetFromZk(eventType, subscription, "0", curator);
+        assertThat(committedOffset, equalTo("25"));
+    }
+
+    private String getCommittedOffsetFromZk(final EventType eventType, final Subscription subscription,
+                                            final String partition, final CuratorFramework curator) throws Exception {
+        final String path = format("/nakadi/subscriptions/{0}/topics/{1}/{2}/offset", subscription.getId(),
```

Comment:
Well, I actually use what @antban designed in his PR. But in general I agree with him, as topics will not hold anything else than partitions it makes sense to skip "partitions" pathNode to save some space.
But if there is theoretical possibility that topic node will hold something else besides partitions, it will make sense to have that "partitions" node to avoid problems in the future.
@antban what is your position on that point?